addons/website_slides/static/src/js/slides_course_page.js

```diff
@@ -0,0 +1,142 @@
+/** @odoo-module **/
+
+import publicWidget from 'web.public.widget';
+import session from 'web.session';
+import  { qweb as QWeb } from 'web.core';
+
+
+/**
+ * Global widget for both fullscreen view and non-fullscreen view of a slide course.
+ * Contains general methods to update the UI elements (progress bar, sidebar...) as well
+ * as method to mark the slide as completed / uncompleted.
+ */
+export const SlideCoursePage = publicWidget.Widget.extend({
+    events: {
+        'click button.o_wslides_button_complete': '_onClickComplete',
+    },
+
+    custom_events: {
+        // The slide has been completed, update the UI
+        'slide_completed': '_onSlideCompleted',
+        // Make a RPC call to complete the slide then update the UI
+        'slide_to_complete': '_onSlideToComplete',
```

Comment:
Perhaps would be better to explain this as doc string for the function, what do you think ?